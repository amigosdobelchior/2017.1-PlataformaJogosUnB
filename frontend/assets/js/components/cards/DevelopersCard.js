import React, {PropTypes} from "react";
import {Card, Popup, Table, Header} from "semantic-ui-react";
import {Link} from "react-router-dom";

export default class DevelopersCard extends React.Component {
    constructor(props){
      super(props);
      this.DEVELOPER = "desenvolvedor";
      this.DESIGN = "design";
      this.MUSICIAN = "musico";
    }
  
    getRoleCells(role){
      const credits = this.props.credits;
      return (<ul>{credits.filter((credit)=> credit.specialty === role)
                    .map((developer) => {
                            if(developer.github_page !== null){
                            return (<Header.Content>
                            <Popup trigger={<Link target="_blank" to={developer.github_page}>{ developer.name }</Link>} 
                                                            content='Link para GitHub'/>
                            </Header.Content>);
                            }
                            else if(developer.email !== null){
                            return(<Header.Content>
                                 <Popup trigger={<div> {developer.name}</div> } content={developer.email}/>
                            </Header.Content>);
                            }
                            else
                                return (<Header.Content>{developer.name }</Header.Content>);

                            })}
      </ul>);
    }

    getTable(role, title){
      return (<Table basic='very' celled collapsing>
              <Table.Header>
                <Table.Row>
                  <Table.HeaderCell>
                    <Header inverted>{title}</Header>
                  </Table.HeaderCell>
                </Table.Row>
              </Table.Header>
              <Table.Body>
                    <Table.Row>
                      <Table.Cell key={role}>
                            {this.getRoleCells(role)}
                      </Table.Cell>
                     </Table.Row>
              </Table.Body>
            </Table>
      );
    }

       render () {

        return (
            <Card fluid>
                <Card.Content header="CrÃ©ditos" />
                <Card.Content>
                  {this.getTable(this.DEVELOPER, "Desenvolvedores")}
                  {this.getTable(this.DESIGN, "Design")}  
                  {this.getTable(this.MUSICIAN, "Musico")}
                </Card.Content>
            </Card>
        );

    }
}

DevelopersCard.propTypes = {
    developer: PropTypes.string.isRequired,
    developers: PropTypes.array.isRequired,
    credits: PropTypes.array.isRequired,
}
