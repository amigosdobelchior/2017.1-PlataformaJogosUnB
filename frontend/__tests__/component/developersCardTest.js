import React, {PropTypes} from 'react';
import DevelopersCard from '../../assets/js/components/cards/DevelopersCard';
import {Card, Popup, Table, Header} from "semantic-ui-react";
import renderer from 'react-test-renderer';
import {Link} from "react-router-dom";

test('Test render DevelopersCard', () => {
  const component = renderer.create(
      <Card fluid>
          <Card.Content header="CrÃ©ditos" />
          <Card.Content>
            "DEVELOPER"
          </Card.Content>
      </Card>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
