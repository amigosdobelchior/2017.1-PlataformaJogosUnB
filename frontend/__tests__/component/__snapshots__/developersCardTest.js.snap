// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render DevelopersCard 1`] = `
<div
  className="ui fluid card"
  href={undefined}
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      Cr√©ditos
    </div>
  </div>
  <div
    className="content"
  >
    DEVELOPER
  </div>
</div>
`;
